### PR TITLE
docs: added poetry related step in tests

### PR DESCRIPTION
**Pull request by ansrivas**
Added the link to the dependencies which should be installed
before running the tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,8 +205,12 @@ postgres=# select * from t;
 ## Running tests
 
 ```sh
+# install poetry if not available on the system.
+# pip install poetry
+
 git clone --recursive https://github.com/neondatabase/neon.git
 make # builds also postgres and installs it to ./tmp_install
+poetry install # install the test dependencies
```
Comment (LizardWizzard) [inline README.md:213]: It shouldnt work without `--no-root` unless newer poetry versions deal with that automatically. There is shortcut script for that: [scripts/pysync](https://github.com/neondatabase/neon/blob/main/scripts/pysync)

Comment (LizardWizzard) [reply README.md:213]: Maybe worth linking this paragraph from earlier part of the readme https://github.com/neondatabase/neon#dependency-installation-notes

Comment (ansrivas) [reply README.md:213]: aha, thanks for pointing it out. 
Somehow I missed the dependency installation notes. I will link it in the PR.

**Review by LizardWizzard — APPROVED** [review]
Thanks!

**Comment by LizardWizzard** [comment]
Some checks have failed because they need to trigger some actions in private repos, and unfortunately this doesnt work for contributors outside neon. The progress of solving this is tracked here https://github.com/neondatabase/neon/issues/1863

Thanks!

**Pull request merged**